File: backend/app/schemas/forecast.py
"""Forecast response schemas."""

from datetime import date
from pydantic import BaseModel, ConfigDict


class ForecastHistoryPoint(BaseModel):
    date: date
    qty: float

    model_config = ConfigDict(from_attributes=True)


class ForecastPredictionPoint(BaseModel):
    date: date
    predicted_qty: float

    model_config = ConfigDict(from_attributes=True)


class MovingAverageForecastResponse(BaseModel):
    product_id: int
    window: int
    days: int
    history: list[ForecastHistoryPoint]
    forecast: list[ForecastPredictionPoint]

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/services/forecast.ts
import { http } from './http';

export interface ForecastHistoryPoint {
    date: string;
    qty: number;
}

export interface ForecastPredictionPoint {
    date: string;
    predicted_qty: number;
}

export interface MovingAverageForecastResponse {
    product_id: number;
    window: number;
    days: number;
    history: ForecastHistoryPoint[];
    forecast: ForecastPredictionPoint[];
}

export async function getMovingAverageForecast(
    productId: number,
    window = 7,
    days = 14
): Promise<MovingAverageForecastResponse> {
    const params = new URLSearchParams({
        product_id: String(productId),
        window: String(window),
        days: String(days),
    });

    return http.get<MovingAverageForecastResponse>(`/forecast/moving-average?${params.toString()}`);
}
